### PR TITLE
Creates the sitemap only for the selected shop in multistore sites.

### PR DESCRIPTION
**Pull request by jagui**
The method `createSitemap($id_shop = 0)` takes a parameter indicating 
the current shop which is never actually passed. As a result of that, 
the module ignores the selected shop in the admin toolbar. 

To further complicate things, if you have a multistore deployement with 
2 shops and 4 langs, the 2 1st langs enabled for shop 1, the remaining 2 
for shop 2, creating the sitemap would create the sub-sitemaps for all 
the 4 langs, which clearly is not the expected outcome.

This patch solves the problem and passes the selected shop to the 
`createSitemap` method.

Fixes https://github.com/PrestaShop/PrestaShop/issues/18649

### CODE REVIEW

File: gsitemap.php [hunk 1]
```diff
@@ -182,11 +182,11 @@ public function getContent()
             }
             Configuration::updateValue('GSITEMAP_DISABLE_LINKS', $meta);
             $this->emptySitemap();
-            $this->createSitemap();
+            $this->createSitemap(Tools::getValue("id_shop"), 0);
```
Comment (Progi1984) [inline gsitemap.php:185]:
```suggestion
            $this->createSitemap(Tools::getValue('id_shop'), 0);
```

File: gsitemap.php [hunk 2]
```diff
@@ -182,11 +182,11 @@ public function getContent()
...
 
         /* If no posted form and the variable [continue] is found in the HTTP request variable keep creating sitemap */
         } elseif (Tools::getValue('continue')) {
-            $this->createSitemap();
+            $this->createSitemap(Tools::getValue("id_shop"), 0);
```
Comment (Progi1984) [inline gsitemap.php:189]:
```suggestion
            $this->createSitemap(Tools::getValue('id_shop'), 0);
```

**Comment by jagui** [comment]
@Progi1984 thanks for the correction. Not sure about the code conventions here, should I squash the patch into a single commit? Cheers, Juan

**Comment by Progi1984** [comment]
@jagui If you can squash, it's better for the history log.

**Comment by jagui** [comment]
@Progi1984 Squashed

File: gsitemap.php (hunk 2 above)
Comment (PierreRambaud) [reply gsitemap.php:189]: Problem not resolved

Comment (Progi1984) [reply gsitemap.php:189]: Ping @jagui

Comment (jagui) [reply gsitemap.php]: sorry folks, updated

**Comment by colinegin** [comment]
Is there an issue for this PR in the general backlog ?

**Comment by marionf** [comment]
Not to my knowledge, @jagui all PR should be linked with an issue, could you please [create an issue](https://github.com/PrestaShop/PrestaShop/issues/new?template=bug_report.md) to describe what is the problem and link it in your PR description's ?

**Comment by jagui** [comment]
@marionf updated with https://github.com/PrestaShop/PrestaShop/issues/18649

**Comment by khouloudbelguith** [comment]
Hi @jagui,

I did not manage to reproduce the issue with PS1.7.6.5 and the `gsitemap` module v4.1.0.
I attached a screen record
https://drive.google.com/file/d/1iyQBf9R2gOTto8cJiQm0mASoNAgf5KTo/view?usp=sharing

Thanks to check and feedback.

**Comment by jagui** [comment]
Hi @khouloudbelguith thanks for the detailed video. 

Perhaps the difference lies on the multi-store configuration: in your video you seem to use [subfolders](http://doc.prestashop.com/display/PS17/Setting+a+shop%27s+URL) for the shops, whereas I'm using different domains. 

Care to recheck? Cheers, Juan

**Comment by khouloudbelguith** [comment]
Hi @jagui,

I tried with different domains and it is ok also.
I attached a second screen record
https://drive.google.com/file/d/1fexMcVsWrHd-uusD1rH01PLyWXRlV53K/view?usp=sharing

Thanks!

**Comment by jagui** [comment]
TL;DR: you're right, I can't repro.

Hi @khouloudbelguith thanks again for the video. Turns out that I submitted the PR from version *v3.2.2* of the module which is the latest one to support PS1.6. In that version I can reproduce the issue and my patch solves it. 

So in case you're interested in keep a maintenance branch for 1.6 I'd be happy to hep. Otherwise I'll close the issue.

Sorry for the inconvenience.

Cheers,
Juan

**Comment by khouloudbelguith** [comment]
@jagui, As you may already know, normal maintenance support for PrestaShop 1.6 ended in October 2018, which is now in [extended maintenance mode until June 30, 2019](https://www.prestashop.com/en/blog/maintenance-extension-prestashop-1-6).

During this period, and in order to ensure the stability of current 1.6 shops, only critical bugs and security fixes will be fixed, that’s why I close it.

Thanks!
